plugin/fzf.vim

```diff
@@ -26,10 +26,62 @@ if exists('g:loaded_fzf')
 endif
 let g:loaded_fzf = 1
 
+function! fzf#shellescape(path)
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set shellslash
+      return '"'.shellescape(a:path).'"'
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return shellescape(a:path)
+endfunction
+
+function! fzf#getcwd()
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set noshellslash
+      return getcwd()
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return getcwd()
+endfunction
+
+function! fzf#fnamemodify(fname, mods)
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set noshellslash
+      return fnamemodify(a:fname, a:mods)
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return fnamemodify(a:fname, a:mods)
+endfunction
+
+function! fzf#expand(fmt)
+  if s:is_win
+    let shellslash = &shellslash
+    try
+      set noshellslash
+      return expand(a:fmt)
+    finally
+      let &shellslash = shellslash
+    endtry
+  endif
+  return expand(a:fmt)
+endfunction
+
```

Comment:
Yes. Ideally, users of the core fzf plugin, such as fzf.vim, should only rely on `fzf#run` and `fzf#wrap` but nothing else, and `fzf#run` should handle all the escaping issues automatically. So I'm not happy that we have to change the implementation of `:FZF` command.

Can we do better? If it's not feasible to properly escape a raw options string for different platforms, we can consider extending `fzf#run` so that it can process array arguments like so:

```vim
call fzf#run({'options': [['--prompt', getcwd()], '--reverse', ['--expect', 'ctrl-a']]})
```